### PR TITLE
LPS-184736 Product QA | Functional Automation - LPS-167476 - Allow developers to retrieve different levels of the many/one to many related elements of a custom object as nestedFields

### PR DESCRIPTION
**Pull request by magjed4289**
https://issues.liferay.com/browse/LPS-184736

### CODE REVIEW

**Comment by magjed4289** [comment]
ci:test:headless-functional:nocompile

**Comment by magjed4289** [comment]
ci:test:relevant

**Comment by magjed4289** [comment]
ci:stop

**Comment by magjed4289** [comment]
ci:test:headless-functional:nocompile

**Comment by magjed4289** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-test/src/testFunctional/tests/NestedFieldsDepth.testcase [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+			ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_plural_label = "subjects",
+				name = "Subject");
+
+			ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_plural_label = "universities",
+				name = "University");
+		}
+
+		task ("And Given studentUsers, studentsUniversities, usersUniversities, studentsSubjects, universityUsers,userStudents, usersSubjects, subjectBooks and booksAuthors relationships created") {
```
Comment (yanan) [inline modules/apps/object/object-rest-test/src/testFunctional/tests/NestedFieldsDepth.testcase:35]: Just a small concern that the relationships in the "And Given" are not exactly same with the ones we use below, maybe we can update "And Given" to include universitySubjects?

**Review by yanan — APPROVED** [review]
Awesome! Thanks for showing the integration test for us to learn!

File: modules/apps/object/object-rest-test/src/testFunctional/tests/NestedFieldsDepth.testcase (hunk 1 above)
Comment (Hijky) [reply modules/apps/object/object-rest-test/src/testFunctional/tests/NestedFieldsDepth.testcase:35]: The same as Ashley, I think we only write the relationship that we created is ok

**Review by Hijky — APPROVED** [review]
The others look good, thanks for showing us the integration test

File: modules/apps/object/object-rest-test/src/testFunctional/tests/NestedFieldsDepth.testcase (hunk 1 above)
Comment (magjed4289) [reply modules/apps/object/object-rest-test/src/testFunctional/tests/NestedFieldsDepth.testcase:35]: yes, thank you guys, I forgot to update this step! 🙏🏻 will modify now :)

**Comment by magjed4289** [comment]
ci:test:relevant

**Comment by magjed4289** [comment]
ci:forward:force

**Comment by magjed4289** [comment]
ci:forward:force

**Pull request closed without merging**